### PR TITLE
refactor redundant code in html5 tech

### PR DESCRIPTION
**Pull request by BrandonOCasey**
## Description
* Refactor any redundant code in the html5 tech to save bytes
* saved 254 bytes in the gzipped bundle, or 0.4% savings

## Specific Changes proposed
* List native properties wrappers at the top and loop through and add them in the constructor
* Better video/audio track code

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by misteroneill** [comment]
Personally, I think this is better and the byte savings are a nice added bonus. 👍 Will review in more detail later.

**Comment by gkatsev** [comment]
We probably should add these functions to the prototype programmatically, not in the constructor.

**Comment by misteroneill** [comment]
Yeah, that sounds like a good idea.

**Comment by dmlap** [comment]
Will we lose documentation for these functions?

**Comment by BrandonOCasey** [comment]
we will lose documentation for these methods

**Comment by BrandonOCasey** [comment]
maybe we can keep the comments without keeping the code?

**Comment by misteroneill** [comment]
There _may_ be a way of documenting them that I'm not aware of (I've tried a few things in the past), but one thing you can definitely do is something like this:

```js
/**
 * Begins playback.
 * 
 * @method play
 * @memberof Player
 */
Player.prototype.play;
```

Not sure that's exactly it, but it's close. I'm not sure if that'd get dropped by uglify or not.

**Comment by misteroneill** [comment]
Another alternative would be to do keep the methods as noops by default:

```js
/**
 * Begins playback.
 */
play() {}
```
Of course, that probably negates the byte savings.

**Comment by pam** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: e00add00aa8192c7e2e517f832d5cba9a9820e59
Build details: https://travis-ci.org/pam/video.js/builds/157173180

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by pam** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: b38b45e4a7e86544b4432bbcb6eeab8dd377e1ca
Build details: https://travis-ci.org/pam/video.js/builds/157174054

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by BrandonOCasey** [comment]
@dmlap @misteroneill added the doc-comments back in for these methods without extra code

**Comment by pam** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 3ec4ccbb93489abab1ad8d4a70ba6489eee1ef90
Build details: https://travis-ci.org/pam/video.js/builds/157190308

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: src/js/tech/html5.js [hunk 1]
```diff
@@ -1418,6 +1019,270 @@ Html5.resetMediaElement = function(el) {
...
 
+/* Native HTML5 element property wrapping ----------------------------------- */
+// Wrap native properties with a getter
+[
+  /**
+   * Paused for html5 tech
+   *
+   * @method Html5.prototype.paused
+   * @return {Boolean}
+   */
```
Comment (misteroneill) [inline src/js/tech/html5.js]: Nice. Glad to see this works! 👍

**Comment by misteroneill** [comment]
LGTM

**Comment by misteroneill** [comment]
LGTM

File: src/js/tech/html5.js [hunk 2]
```diff
@@ -417,24 +389,8 @@ class Html5 extends Tech {
...
-    const techTracks = this.videoTracks();
-    const elTracks = this.el().videoTracks;
-
-    this.removeOldTracks_(techTracks, elTracks);
-  }
-
   /**
    * Play for html5 tech
-   *
-   * @method play
```
Comment (gkatsev) [inline src/js/tech/html5.js]: does jsdoc figure out this `@method` itself?

File: src/js/tech/html5.js [hunk 3]
```diff
@@ -1418,6 +1019,270 @@ Html5.resetMediaElement = function(el) {
...
+  'loop',
+  /**
+   * Set desired speed at which the media resource is to play
+   *
+   * @param {Number} val Speed at which the media resource is to play
+   * @method Html5.prototype.setPlaybackRate
+   */
+  'playbackRate'
+].forEach(function(prop) {
+  Html5.prototype[`set${toTitleCase(prop)}`] = function(v) {
```
Comment (gkatsev) [inline src/js/tech/html5.js]: might be more readable to just do `'set' + toTitleCase(prop)`.

File: src/js/tech/html5.js (hunk 2 above)
Comment (misteroneill) [inline src/js/tech/html5.js]: I believe that is the case.

Comment (BrandonOCasey) [inline src/js/tech/html5.js]: yeah it does

File: src/js/tech/html5.js (hunk 3 above)
Comment (BrandonOCasey) [inline src/js/tech/html5.js]: done

**Pull request merged**